internal/authentication/azure/auth.go

```diff
@@ -42,30 +44,37 @@ func NewEnvironmentSettings(resourceName string, values map[string]string) (Envi
 		return es, err
 	}
 	es.AzureEnvironment = azureEnv
-	switch resourceName {
-	case "azure":
+	switch {
+	case resourceName == "azure":
 		// Azure Resource Manager (management plane)
 		es.Resource = azureEnv.TokenAudience
-	case "keyvault":
+	case resourceName == "keyvault":
 		// Azure Key Vault (data plane)
 		es.Resource = azureEnv.ResourceIdentifiers.KeyVault
-	case "storage":
+	case resourceName == "storage":
 		// Azure Storage (data plane)
 		es.Resource = azureEnv.ResourceIdentifiers.Storage
-	case "cosmosdb":
+	case resourceName == "cosmosdb":
 		// Azure Cosmos DB (data plane)
 		es.Resource = azureEnv.ResourceIdentifiers.CosmosDB
-	case "servicebus":
+	case resourceName == "servicebus":
 		es.Resource = azureEnv.ResourceIdentifiers.ServiceBus
-	case "eventhubs":
+	case resourceName == "eventhubs":
 		// Azure EventHubs (data plane)
 		// For documentation https://docs.microsoft.com/en-us/azure/event-hubs/authorize-access-azure-active-directory#overview
 		// The resource name to request a token is https://eventhubs.azure.net/, and it's the same for all clouds/tenants.
 		// Kafka connection does not factor in here.
 		es.Resource = "https://eventhubs.azure.net"
-	case "signalr":
+	case resourceName == "signalr":
 		// Azure SignalR (data plane)
 		es.Resource = "https://signalr.azure.com"
+	case strings.HasSuffix(resourceName, "azconfig.io"):
```

Comment:
Please don't change this switch to look for suffixes. The idea of `resourceName` is that it should be a constant string such as `eventhubs`, `storage`, etc (perhaps we should make them real constants instead). Define a constant such as `"appconfig"` and use that.

Reason is that suffixes such as `azconfig.io` may be different in other sovereign clouds

Reply:
I have to run some test to check if asking for a token for resource `https://azconfig.io` will make the cut here.

Reply:
It does!!! 😎